go/gilded-rose_test.go

```diff
@@ -2,14 +2,60 @@ package main
 
 import "testing"
 
-func Test_Foo(t *testing.T) {
-	var items = []*Item{
-		&Item{"foo", 0, 0},
+// Expected is an array of integers representing expected sellIn and quality
+type Expected [2]int
+
+// An ItemTest is a struct which includes an Item along with the expected values
+// for sellIn and quality after the quality is updated by the UpdateQuality function
+type ItemTest struct {
+	*Item
+	Expected
+}
+
+// GetItemTestsComponents returns an slice of Items and Expecteds from a slice of
+// ItemTests. This is useful so that we can have our tests as ItemTests to keep the expected
+// results along with the Item definitions. This function can be used to extract the slice of Items
+// which can then be passed to the UpdateQuality function
+func GetItemTestsComponents(itemTests []ItemTest) ([]*Item, []Expected) {
+	var items []*Item
+	var expected []Expected
+
+	for _, itemTest := range itemTests {
+		items = append(items, itemTest.Item)
+		expected = append(expected, itemTest.Expected)
+	}
+
+	return items, expected
+}
+
+func Test_Items(t *testing.T) {
+	var itemTests = []ItemTest{
```

Comment:
Test cases are only covering 1 day change, I think it would be worth to include longer periods as well.

Reply:
Yes. I have seen other people test multiple days but I am not convinced on any reasons for doing this. What are we testing? The quality is still updated day by day. This could be tested by well chosen single test cases.

Reply:
You are right, when thinking about it now, it wouldn't make much difference, maybe just you could cover more code in a single test case.